From b687f4ede4c620854d940a9f788971b1ae32195c Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Fri, 26 Sep 2014 18:12:45 +0800
Subject: [PATCH] qt-apps: applicationlauncher: adapter to the screen's
 resolution

If it's 480x272 then launch the small resolution version.
Otherwise launch the general version (800x480).

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 launch.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/launch.sh b/launch.sh
index 70b6fdb..a477311 100755
--- a/launch.sh
+++ b/launch.sh
@@ -3,4 +3,14 @@
 # when eval returns.
 
 eval "$@"
-/opt/ApplicationLauncher/ApplicationLauncher.sh
+
+resolution=`fbset | grep mode | awk -F '"' '{print $2}' | awk -F "-" '{print $1}'`
+
+case "$resolution" in
+	"480x272")
+	 	/opt/ApplicationLauncher/ApplicationLauncher-small.sh
+	;;
+	*)
+		/opt/ApplicationLauncher/ApplicationLauncher.sh
+	;;
+esac
-- 
1.9.1

